### PR TITLE
Resolves SmesVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the SmesVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Power/SMES/SmesVisualizerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            return;
+
+        sprite.LayerMapSet(Layers.Input, sprite.AddLayerState("smes-oc0"));
+        sprite.LayerSetShader(Layers.Input, "unshaded");
+        sprite.LayerMapSet(Layers.Charge, sprite.AddLayerState("smes-og1"));
+        sprite.LayerSetShader(Layers.Charge, "unshaded");
+        sprite.LayerSetVisible(Layers.Charge, false);
+        sprite.LayerMapSet(Layers.Output, sprite.AddLayerState("smes-op0"));
+        sprite.LayerSetShader(Layers.Output, "unshaded");
+    }
```
Comment (AJCM-git) [inline Content.Client/Power/SMES/SmesVisualizerSystem.cs:27]: kill by mapping layers accordingly

File: Content.Client/Power/SMES/SmesVisualizerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+﻿namespace Content.Client.Power.SMES;
+
+[RegisterComponent]
+public sealed class SmesVisualizerComponent : Component
+{}
```
Comment (AJCM-git) [inline Content.Client/Power/SMES/SmesVisualizerComponent.cs:5]: visuals, make { } 
{
}

File: Content.Client/Power/SMES/SmesVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Power.SMES;
+
+public sealed class SmesVisualizerSystem : VisualizerSystem<SmesVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SmesVisualizerComponent, ComponentInit>(OnInit);
```
Comment (AJCM-git) [inline Content.Client/Power/SMES/SmesVisualizerSystem.cs:12]: oncomponentinit

File: Content.Client/Power/SMES/SmesVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        else
+        {
+            args.Sprite.LayerSetState(Layers.Input, "smes-oc0");
+            args.Sprite.LayerSetState(Layers.Output, "smes-op1");
+        }
+    }
+}
+
+enum Layers : byte
```
Comment (AJCM-git) [inline Content.Client/Power/SMES/SmesVisualizerSystem.cs:70]: rename to SmesVisualLayers

File: Content.Client/Power/SMES/SmesVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                    args.Sprite.LayerSetState(Layers.Input, "smes-oc0");
+                    args.Sprite.LayerSetState(Layers.Output, "smes-op2");
+                    break;
+            }
+        }
+        else
+        {
+            args.Sprite.LayerSetState(Layers.Input, "smes-oc0");
+            args.Sprite.LayerSetState(Layers.Output, "smes-op1");
+        }
```
Comment (AJCM-git) [inline Content.Client/Power/SMES/SmesVisualizerSystem.cs:66]: unharcode

**Review by AJCM-git — DISMISSED** [review]

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

**Comment by TemporalOroboros** [comment]
@DrSmugleaf Commits tested and pushed. Reopen whenever's convenient.

File: Content.Client/Power/SMES/SmesVisualizerComponent.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿namespace Content.Client.Power.SMES;
+
+[RegisterComponent]
+public sealed class SmesVisualsComponent : Component
+{
```
Comment (metalgearsloth) [inline Content.Client/Power/SMES/SmesVisualizerComponent.cs:5]: Same as APC, should probably use SmesComponent

Comment (AJCM-git) [reply Content.Client/Power/SMES/SmesVisualizerComponent.cs:5]: Moving smescomponent to shared is way more doable in this case than apc component so you should give it a shot

**Pull request merged**